src/lib/insights/listener.tsx

```diff
@@ -23,19 +26,50 @@ const findInsightsTarget = (
   return element;
 };
 
+const parseInsightsEvent = element => {
+  const serializedPayload = element.getAttribute('data-insights-event');
+
+  if (typeof serializedPayload !== 'string') {
+    throw new Error(
+      'The insights middleware expects `data-insights-event` to be a base64-encoded JSON string.'
+    );
+  }
+
+  try {
+    return JSON.parse(atob(serializedPayload));
+  } catch (error) {
+    throw new Error(
+      'The insights middleware was unable to parse `data-insights-event`.'
+    );
+  }
+};
+
 const insightsListener = (BaseComponent: any) => {
   function WithInsightsListener(props: WithInsightsListenerProps) {
     const handleClick = (event: MouseEvent): void => {
+      // old way, e.g. instantsearch.insights("clickedObjectIDsAfterSearch", { .. })
```

Comment:
We now allow people to use the two at the same time? Shouldn't we check the new way, and then the old way exclusively?

```js
if (props.sendEvent) {}
else {}
```

Reply:
Makes sense! https://github.com/algolia/instantsearch.js/pull/4456/commits/61ca509ced8082c6efd007d8ab04f0c65645bba6